router/pkg/metric/metric_store.go

```diff
@@ -141,6 +156,19 @@ func NewStore(opts ...Option) (Store, error) {
 	return h, nil
 }
 
+// setCardinalityLimit sets the cardinality limit for open telemetry.
+// This feature is experimental in otel-go and may be exposed in a different way in the future.
+// In order to avoid creating a large number of metric streams, we set a hard limit that can be collected for a single instrument.
+func setCardinalityLimit(limit int) error {
+	if limit < 0 {
+		// setting a limit of 0 disables  the cardinality limit
+		limit = 0
+	}
+
+	val := strconv.Itoa(limit)
+	return os.Setenv("OTEL_GO_X_CARDINALITY_LIMIT", val)
```

Comment:
This is fine but it is a potential race because setting an env within a test will effects all other tests. We have to write those tests in a separate file and not in parallel.

Reply:
Good catch